src/js/tracks/text-track-settings.js

```diff
@@ -290,21 +305,28 @@ class TextTrackSettings extends Component {
    *         The DOM element that gets created.
    * @private
    */
-  createElSelect_(key) {
+  createElSelect_(key, legendId = '') {
     const config = selectConfigs[key];
     const id = config.id.replace('%s', this.id_);
 
     return [
       createEl('label', {
+        id,
         className: 'vjs-label',
         textContent: this.localize(config.label)
       }, {
-        for: id
       }),
-      createEl('select', {id}, undefined, config.options.map(o => {
+      createEl('select', {}, {
+        'aria-labeledby': `${legendId} ${id}`
+      }, config.options.map(o => {
+        const optionId = id + '-' + o[1];
+
         return createEl('option', {
+          id: optionId,
           textContent: this.localize(o[1]),
           value: o[0]
+        }, {
+          'aria-labeledby': `${legendId} ${id} ${optionId}`
```

Comment:
I was testing with voiceover on mac. Seems like Safari reads it out like `White, Text Color, menu item` but on chrome it reads it out only as `white` unless I add this `labelledby`.